osmomath/decimal_test.go

```diff
@@ -30,18 +30,105 @@ func TestDecimalTestSuite(t *testing.T) {
 	suite.Run(t, new(decimalTestSuite))
 }
 
+// BigIntDecorator is implemented by osmomath's numerical values, as Dec, BigDec, Int, Uint
+type BigIntDecorator interface {
+	BigInt() *big.Int
+}
+
+// Precision returns a precision of decimal's underlying implementation
+func Precision(decimal interface{}) int64 {
+	switch any(decimal).(type) {
+	case osmomath.BigDec:
+		return osmomath.PrecisionBigDec
+	case osmomath.Dec:
+		return osmomath.PrecisionDec
+	case osmomath.Int, osmomath.Uint:
+		return 0
+	default:
+		panic(fmt.Sprintf("%T type is not a decimal", decimal))
+	}
+}
```

Comment:
question to reviewers: should this function be exported in osmomath package? is there a use case for it somewhere else in the repo?

Reply:
Not as I am concerned of for now. I think we can leave this here for now